Lib/logging/__init__.py

```diff
@@ -793,6 +798,9 @@ def createLock(self):
         Acquire a thread lock for serializing access to the underlying I/O.
         """
         self.lock = threading.RLock()
+        os.register_at_fork(before=self.acquire,
```

Comment:
This may register an arbitrary number of callbacks across an application's lifetime, also it makes the handles eternal by keeping a reference to all of them. Instead, we could have a global WeakSet of handlers and a single set of callbacks that iterate over the live handlers.

Reply:
Another option would be a `before=lambda acquire=weakref.WeakMethod(self.acquire): acquire()()`... I'm used to applications that don't go creating tons of `logging.Handler` instances and throwing them away all over the place. :)

I think I'll go with the WeakSet approach to avoid an every growing list of callables in that strange Handler discarding application case...